python/cugraph/tests/test_maximum_spanning_tree.py

```diff
@@ -0,0 +1,140 @@
+# Copyright (c) 2019-2020, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import gc
+
+import pytest
+
+import cugraph
+from cugraph.tests import utils
+from cugraph.utilities import check_nx_graph
+from cugraph.utilities import cugraph_to_nx
+import rmm
+import cudf
+import time
+import numpy as np
+import pandas as pd
+
+# Temporarily suppress warnings till networkX fixes deprecation warnings
+# (Using or importing the ABCs from 'collections' instead of from
+# 'collections.abc' is deprecated, and in 3.8 it will stop working) for
+# python 3.7.  Also, this import networkx needs to be relocated in the
+# third-party group once this gets fixed.
+import warnings
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=DeprecationWarning)
+    import networkx as nx
+
+print("Networkx version : {} ".format(nx.__version__))
+
+
+def compare_mst(mst_cugraph, mst_nx):
+    mst_nx_df = nx.to_pandas_edgelist(mst_nx)
+    edgelist_df = mst_cugraph.view_edge_list()
+    assert len(mst_nx_df) == len(edgelist_df)
+
+    # check cycles
+    Gnx = nx.from_pandas_edgelist(
+        edgelist_df.to_pandas(),
+        create_using=nx.Graph(),
+        source="src",
+        target="dst",
+    )
+    try:
+        l = find_cycle(G, source=None, orientation="ignore")
+        print(l)
+    except:
+        pass
+
+    # check total weight
+    cg_sum = edgelist_df["weights"].sum()
+    nx_sum = mst_nx_df["weight"].sum()
+    print(cg_sum)
+    print(nx_sum)
+    assert np.isclose(cg_sum, nx_sum)
+
+
+@pytest.mark.parametrize("graph_file", utils.DATASETS_UNDIRECTED_WEIGHTS)
+def test_maximum_spanning_tree_nx(graph_file):
```

Comment:
This and the below test should be combined with the tests from `test_minimum_spanning_tree.py`. Tests can take parameter like:
```
@pytest.mark.parametrize("spanning_tree_class", ["maximum", "minimum"])
```
Then, if it's type maximum, then just negate the weights

Reply:
Not sure about that. The goal here is to test `maximum_spanning_tree` feature. 
cuGraph does negate the weights itself actually as part of the feature but doing it manually in the test would defeat the purpose of testing the feature. Also, we need to compare against Nx's `maximum_spanning_tree`. As a result, notice that merging would result in branches all over in the tests.